### PR TITLE
ARROW-484: Revise README to include more detail about software components

### PR DESCRIPTION
**Pull request by wesm**
Also closes #14.

### CODE REVIEW

**Comment by wesm** [comment]
@xhochy or @julienledem could you give a quick review and let me know if I should add or change anything else? I will generalize the note about binary format since we're adding support for other styles of wire protocols

File: README.md [hunk 1]
```diff
@@ -32,17 +32,38 @@ Arrow is a set of technologies that enable big-data systems to process and move
...
-Arrow is an [Apache Software Foundation](www.apache.org) project. More info can be found at [arrow.apache.org](http://arrow.apache.org).
+Arrow is an [Apache Software Foundation](www.apache.org) project. Learn more at
+[arrow.apache.org](http://arrow.apache.org).
+
+#### What's in the Arrow libraries?
+
+The reference Arrow implementations contain a number of distinct software
+components:
+
+- Columnar vector/array and table row batch containers supporting nested data
```
Comment (xhochy) [inline README.md]: Maybe we should drop `DataFrame` (or `superset of a DataFrame`) somewhere. That's at least what people in the PyData space are more used too.

Comment (wesm) [inline README.md]: done

**Comment by wesm** [comment]
+1

**Pull request closed without merging**